packages/metro/src/DeltaBundler/traverseDependencies.js

```diff
@@ -309,12 +312,71 @@ async function addDependency<T>(
   options.onDependencyAdded();
 }
 
-function removeDependency<T>(
+/**
+ * Recursively look up `inverseDependencies` until it is empty,
+ * returning a set of paths for the last module that does not have
+ * `inverseDependencies`.
+ */
+function getAllTopLevelInverseDependencies<T>(
+  inverseDependencies: Set<string>,
+  parentModule: string,
+  graph: Graph<T>,
+  currModule: string,
+  visited: Set<string>,
+): Set<string> {
+  if (visited.has(currModule)) {
+    return new Set();
+  }
+  visited.add(currModule);
+  if (!inverseDependencies.size || currModule === parentModule) {
+    return new Set([currModule]);
+  }
+
+  return Array.from(inverseDependencies)
+    .filter(inverseDep => graph.dependencies.has(inverseDep))
+    .reduce((acc, inverseDep) => {
+      const mod = graph.dependencies.get(inverseDep);
+      if (!mod) {
+        return acc;
+      }
+      getAllTopLevelInverseDependencies(
+        mod.inverseDependencies,
+        parentModule,
+        graph,
+        inverseDep,
+        visited,
+      ).forEach(x => {
+        acc.add(x);
+      });
+      return acc;
+    }, new Set());
+}
+
+/**
+ * Given `inverseDependencies`, tracing back inverse dependencies to
+ * see if it only leads back to `parentModule`.
+ */
+async function canSafelyRemoveFromParentModule<T>(
```

Comment:
I think this function is synchronous? In which case we should remove `async`

Reply:
Ahh, nice catch :)